### ISSUE TITLE
[New rule] avoid named from keywords list

### ISSUE THREAD

**Issue by andrewst** [dart-code-checker/dart-code-metrics#136]
**Please describe what the rule should do:**
avoid named from keywords list https://dart.dev/guides/language/language-tour#keywords
**If your rule is inspired by other please provide link to it:**

**What category of rule is this? (place an "X" next to just one item)**

[ X] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

```dart
void main() {

  const abstract = 123;
  const import = 123;
  const as = 123;
  const export = 123;
  const interface = 123;
  const sync = 123;
  const async = 123;
  const await = 123;
  const extension = 123;
  const library = 123;
  const external = 123;
  const mixin = 123;
  const factory = 123;
  const typedef = 123;
  const on = 123;
  const operator = 123;
  const part = 123;
  const covariant = 123;
  const Function = 123;
  const get = 123;
  const yield = 123;
  const deferred = 123;
  const hide = 123;
  const set = 123;
  const show = 123;
  const dynamic = 123;
  const implements = 123;
  const static = 123;
}
```

**Are you willing to submit a pull request to implement this rule?**
may be

**Comment by incendial** [comment]
We decided not to implement it due to a small impact and potential usage.
